### ISSUE TITLE
apt-get autoremove (old kernels)

### ISSUE THREAD

**Issue by Commifreak** [DE-IBH/apt-dater#102]
Title says it all :)

I would like a feature ti run an autoremove through all hosts. Sometimes the default boot-partition is out of space.. ;)

Regards,

**Comment by DejavuDmitry** [comment]
Yes. Please add this feature or feature that add some function to set custom hot-keys to run my own command on machines.

**Comment by kiranos** [comment]
I'd like this feature aswell, would be really nice to have it as an option.

**Comment by janvonde** [comment]
+1

**Comment by AikedeJongste** [comment]
Yes please!

**Comment by phaidros7** [comment]
:+1:

**Comment by moonmeister** [comment]
+1

**Comment by chrislp** [comment]
+1

**Comment by MikeLund** [comment]
In app-dater.conf:

> [Commands]
> CmdRefresh=apt-dater-host refresh
> CmdUpgrade=**apt-get autoremove && apt-get autoclean &&** apt-dater-host upgrade
> CmdInstall=apt-dater-host install %s

Maybe not super-smart, but has worked fine for me for at least 6 months now.

**Comment by noudAndi** [comment]
Does anybody know, where to put this config in the new .xml style files?

**Comment by DejavuDmitry** [comment]
@noudAndi maybe
```
$ sed -n 109,112p /usr/lib/x86_64-linux-gnu/apt-dater/cmd
handle_generic_ssh () {
    AD_CMD_REFRESH="apt-dater-host refresh"
    AD_CMD_UPGRADE="apt-dater-host upgrade"
    AD_CMD_INSTALL="apt-dater-host install %s"
```
?

**Comment by mathiashro** [comment]
+1

**Comment by hatclub** [comment]
Having had yet another VM get it's package manager in a twist because of the boot partition filling up, I discovered the apt-dater hooks functionality introduced since v0.8.1.

I have resorted to placing, on the machine I run apt-dater from, chmod 700, /etc/apt-dater/post-upg.d/autoremove with

```
#!/bin/bash
if [ -z "$AD_SSH_PORT" ]; then
        AD_SSH_PORT=22
fi
ssh $AD_SSH_USER@$AD_SSH_HOST -p $AD_SSH_PORT ". /etc/os-release; case \$ID in debian|ubuntu|mint) sudo apt-get autoremove; ;; fedora|rhel|centos) sudo yum autoremove; ;; *) echo Autoremove not supported for OS \$ID; ;; esac"

```

It's not the most elegant solution, but it gets the job done and will run an autoremove pass after any upgrade command runs. 

You can adjust as you see fit the cases / commands for your environment.
